app/services/host_counter.rb

```diff
@@ -28,14 +28,16 @@ def cache
 
   def counted_hosts
     hosts_scope = Host::Managed.reorder('')
+    grouping = "#{@association}_id"
     case @association.to_s
     when 'organization', 'location'
       # If we are on /organizations or /locations, this allows to display the
       # count for hosts not in the current organization & location.
       hosts_scope = hosts_scope.unscoped
     when 'subnet', 'domain'
       hosts_scope = hosts_scope.joins(:primary_interface)
+      grouping = grouping.prepend("#{Nic::Base.table_name}.")
```

Comment:
This assignment might not be needed, `.prepend()` modifies in place.

Reply:
```suggestion
      grouping.prepend("#{Nic::Base.table_name}.")
```
It's indeed not needed.